presto-main/src/main/java/com/facebook/presto/sql/planner/LocalExecutionPlanner.java

```diff
@@ -2323,13 +2332,22 @@ public PhysicalOperation visitTableWriter(TableWriterNode node, LocalExecutionPl
                     statisticsAggregation,
                     getVariableTypes(node.getOutputVariables()),
                     tableCommitContextCodec,
-                    stageExecutionDescriptor.isRecoverableGroupedExecution()
-                            ? PageSinkCommitStrategy.LIFESPAN_COMMIT
-                            : PageSinkCommitStrategy.NO_COMMIT);
+                    getPageSinkCommitStrategy());
 
             return new PhysicalOperation(operatorFactory, outputMapping.build(), context, source);
         }
 
+        private PageSinkCommitStrategy getPageSinkCommitStrategy()
+        {
+            if (pageSinkCommitRequired) {
+                return TASK_COMMIT;
+            }
+            if (stageExecutionDescriptor.isRecoverableGroupedExecution()) {
```

Comment:
Should we first check whether `isRecoverableGroupedExecution` is `true` and return `LIFESPAN_COMMIT` first? I feel like `isRecoverableGroupedExecution` is a stronger predicate than `pageSinkCommitRequired`. What do you think?

Reply:
I agree. Let me do that.